receiver/prometheusreceiver/internal/prom_to_otlp.go

```diff
@@ -21,11 +21,24 @@ import (
 	conventions "go.opentelemetry.io/collector/model/semconv/v1.5.0"
 )
 
+const (
+	localHostIPv4 = "127.0.0.1"
+	anyIPIPv4     = "0.0.0.0"
+)
+
+func sanitizeHost(host string) string {
+	if host == anyIPIPv4 {
+		return localHostIPv4
+	}
+	return host
+}
+
 func createNodeAndResourcePdata(job, instance, scheme string) pdata.Resource {
 	host, port, err := net.SplitHostPort(instance)
 	if err != nil {
 		host = instance
 	}
+	host = sanitizeHost(host)
```

Comment:
Wouldn't it be better to **remove** the host name *Attribute* when its value is `0.0.0.0`, `localhost`, `127.0.0.1` or `::1` (or any variation of this) and let the [resourcedetection processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/resourcedetectionprocessor) do the job or setting this attribute properly (with the actual host name of the system).

Because eventually, the metric is going to be stored somewhere else, where *localhost* will be either meaningless or misleading.

Reply:
I certainly would prefer that (e.g. the way we as a vendor use `host.name` makes it so `localhost` and similar attributes are practically useless), but I was assuming these attributes were necessary for something Prometheus-related.

Maybe @dashpole can comment on this?

Reply:
If you are using the resource detection processor, it overrides the existing values by default anyways.  I double-checked, and host.name isn't used for anything prometheus-related

Reply:
> If you are using the resource detection processor, it overrides the existing values by default anyways. 

There are valid use cases where you don't want the `resourcedetectionprocessor` to override the hostname (e.g. chaining multiple Collectors), or maybe you can't/don't want to use it at all (e.g. your Collector distro doesn't support it or you don't want the performance hit).

>  I double-checked, and host.name isn't used for anything prometheus-related

Personally I would prefer removing it then, if everyone is on board with this. I agree with @bertysentry that having a localhost-like value for `host.name` is, while spec-compliant, not very useful in practice.

Reply:
Removing it only when its a localhost equivalent? Or removing it entirely?

Reply:
**resourcedetectionprocessor** has an `override` option that can be set to `false` to prevent it from overwriting attributes that are already set.

Removing the `host.name` attributes in **prometheusreceiver** when its value is `localhost` (or any variation of localhost, incl. `0.0.0.0`) will then work very well in combination with **resourcedetectionprocessor** (and `override: false`). 

Typical example: the internal *otelcol* Prometheus exporter, whose metrics are currently attached to `0.0.0.0`, which are not usable once these metrics from several collectors are aggregated into a Prometheus server.

Reply:
> Removing it only when its a localhost equivalent?

Only when localhost or equivalent, yes